File: user-service/src/main/java/rs/edu/raf/dto/GenerateCodeDTO.java
package rs.edu.raf.dto;

import jakarta.validation.constraints.Pattern;

public record GenerateCodeDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    String email
) { }

File: user-service-kotlin/src/main/kotlin/org/example/userservicekotlin/dto/GenerateCodeDTO.kt
package org.example.userservicekotlin.dto

import jakarta.validation.constraints.Pattern

data class GenerateCodeDTO(
    @Pattern(regexp = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$", message = "Email address must be valid!")
    val email: String
)
